### DISCUSSION TITLE
Database generated default timestamp

### DISCUSSION THREAD

**Discussion by kylejw2** [prisma/prisma#21547] [Q&A] [answered]
### Question

I'm building an app that offers a free trial experience. I've got a column in my database called `freeTrialExpiresAt` which refers to when the company's free trial expires. In my `schema.prisma` file, I have the following line:
```
freeTrialExpiresAt       DateTime                @default(dbgenerated("NOW() + INTERVAL '30 days'"))
```

What I'm trying to accomplish is have a dynamically generated timestamp. Whenever the company's record is created, the database automatically calculates the `freeTrialExpiresAt` by adding 30 days to the current timestamp. The problem I'm running into is that every time I add a new migration, Prisma thinks I want to recalculate this default timestamp. So each subsequent migration includes:
```
-- AlterTable
ALTER TABLE "Company" ALTER COLUMN "freeTrialExpiresAt" SET DEFAULT NOW() + INTERVAL '30 days';
```

I know I could remove the default timestamp and manage it in my application code, but is anyone aware of how I could let Prisma and my postgres db manage this default value?

### How to reproduce (optional)

1. Create a new schema.prisma file
2. Create a new model in the schema.prisma file.
3. Add a DateTime column such as `freeTrialExpiresAt       DateTime                @default(dbgenerated("NOW() + INTERVAL '30 days'"))` to the model.
4. Run `npx prisma migrate dev` to generate and run the migration.
5. Run `npx prisma migrate dev --create-only` and notice the new migration file wants to alter the timestamp to recalculate it.

### Expected behavior (optional)

I expected Prisma to stop including the ALTER COLUMN command for a default value that should be managed by the database and not on every migration.

### Information about Prisma Schema, Client Queries and Environment (optional)

- OS: macOS
- Database: Postgres
- Node.js version: v18.9.0
- Prisma version: 4.13.0

**Comment by ludralph** [comment, marked as answer]
Hi @kylejw2 👋 

Thank you for raising this question.

You're encountering a known issue where Prisma generates unnecessary migrations for default values, especially when using shorthand interval definitions in Postgres `dbgenerated` defaults. This issue has been reported here:

- https://github.com/prisma/prisma/issues/11412

As a workaround, you can use the expanded definition of the interval in your Prisma schema. Instead of using `'30 days'`, you can use `'30 days'::interval`. Here's how you can modify your schema:
```prisma
model Token {
  id Int @default(autoincrement()) @id
  freeTrialExpiresAt DateTime @default(dbgenerated("(NOW() + '30 days'::interval)"))
}
```

This should prevent Prisma from generating a new migration every time you run `prisma migrate dev`.


If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by kylejw2** [reply]
Worked like a charm. Thank you @ludralph !

**Reply by uxinnuendo** [reply]
Since this wasn't titled for Postgres specifically; for anyone using MYSQL, at the time of writing this, the migration drift issue is still unresolved (and postgres workarounds don't work on mysql).

eg. @default(dbgenerated("(CURRENT_TIMESTAMP + INTERVAL 1 HOUR)"))

**Reply by JackMBurch** [reply]
Confirming, doesn't seem to be a way to handle this with postgres at the moment.
